reflex/components/component.py

```diff
@@ -66,6 +66,10 @@ class Component(Base, ABC):
 
     # components that cannot be children
     invalid_children: List[str] = []
+
+    # components that are only allowed as children
+    valid_children: List[str] = []
```

Comment:
I'm also thinking in a follow up if these should maybe be `List[Type[Component]]` rather than `str`